src/main/java/de/tum/in/www1/artemis/web/rest/ModelingAssessmentResource.java

```diff
@@ -249,9 +250,9 @@ public ModelingAssessmentResource(AuthorizationCheckService authCheckService, Us
             compassService.addAssessment(exerciseId, submissionId, result.getFeedbacks());
         }
 
-        // remove circular dependencies
-        if (result.getParticipation() != null && result.getParticipation().getResults() != null) {
-            result.getParticipation().getResults().stream().forEach(participationResult -> participationResult.setParticipation(null));
+        // remove circular dependencies if the results of the participation are there
+        if (result.getParticipation() != null && Hibernate.isInitialized(result.getParticipation().getResults()) && result.getParticipation().getResults() != null) {
+            result.getParticipation().getResults().forEach(participationResult -> participationResult.setParticipation(null));
```

Comment:
I get that we want to send the result with its participation, but why also attach all other results to the participation?
This seems weird to me and like the real issue here.
What about: `result.getParticipation().setResults(null)`

Reply:
In most cases this will not make a difference as the results collection seems to be a Hibernate proxy anyway. Additionally, we only have one result per participation for modeling exercises.
However, you are right that sending the result collection is not necessary, I changed it to `result.getParticipation().setResults(null)`.

Reply:
:+1: